### PR TITLE
[VC-36032] Set User-Agent header containing the agent version in all HTTP requests

### PR DESCRIPTION
**Pull request by wallrj**
In [VC-36032](https://venafi.atlassian.net/browse/VC-36032) we aim to provide:
> Enhanced troubleshooting through improved logging

By adding user-agent headers to *all* HTTP requests, it will allow the Venafi platform team to know which version of the agent is making requests to the resource and auth APIs, by analysing the request headers among the server logs.

I tried writing tests for this, but the it would have required some significant refactoring of the client code,
for example there's no easy way to intercept the  Auth0 client request made by the Jetstack-Secure client in client_oauth.go.
That's a bit of a lame excuse, but I would like to propose that we refactor the Venafi client code in a followup PR and at the same time add tests which verify that all the flavours of Venafi client do include user agent headers.

[VC-36032]: https://venafi.atlassian.net/browse/VC-36032?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pkg/agent/config_test.go [hunk 1]
```diff
@@ -178,7 +178,7 @@ func Test_ValidateAndCombineConfig(t *testing.T) {
 
 		// The log line printed by pflag is not captured by the log recorder.
 		assert.Equal(t, testutil.Undent(`
-			INFO Using the Jetstack Secure OAuth auth mode since --credentials-file was specified without --venafi-cloud.
+			INFO Authentication mode mode="Jetstack Secure OAuth" reason="--credentials-file was specified without --venafi-cloud"
```
Comment (wallrj) [inline pkg/agent/config_test.go:181]: Another drive-by fix. This test was introduced in #628 but #625 which altered the logging format (and which was merged afterwards) had not been rebased and did not alter the expected log output in this test.

**Review by maelvls — APPROVED** [review]
Thanks for fixing this, I wish we had done it sooner. I've realized the importance of the user agent header when responding to last weeks venctl/agent outage.

I haven't tested the PR, but I'm very confident that nothing will break. Nothing in the PR could break anything, I'd say.

File: pkg/agent/config_test.go (hunk 1 above)
Comment (wallrj) [reply pkg/agent/config_test.go:181]: The tests are not being run on the master branch and the green badge on the README file does not reflect the status of the current branch:
 * https://github.com/jetstack/jetstack-secure/actions/workflows/tests.yaml?query=branch%3Amaster
 * https://github.com/jetstack/jetstack-secure/blob/master/README.md
